evalml/pipelines/time_series_regression_pipeline.py

```diff
@@ -95,6 +95,8 @@ def predict(self, X, y=None, objective=None):
         if self.estimator.predict_uses_y:
             y_arg = y
         predictions = self.estimator.predict(features_no_nan, y_arg).to_series()
+        predictions.index = y.index
+        predictions = self.inverse_transform(predictions).to_series()
```

Comment:
Right now the time series regression pipeline is the only pipeline that uses the `inverse_transform` method. 

As we add more target transformers (target encoder as a component, https://github.com/alteryx/evalml/issues/914), I expect the other pipeline classes to call the `inverse_transform` method.